src/main/java/eggventory/commands/add/AddStockCommand.java

```diff
@@ -48,17 +48,16 @@ public String execute(StockList list, Cli cli, Storage storage) {
         if (list.isExistingStockCode(stockCode)) {
             output = String.format("Sorry, the stock code \"%s\" is already assigned to a stock in the system. "
                     + "Please enter a different stock code.", stockCode);
-            cli.print(output);
-            return output;
 
         } else {
             list.addStock(stockType, stockCode, quantity, description);
             storage.save(list);
             output = String.format("Nice! I have successfully added the stock: StockType: %s StockCode: %s "
                     + "Quantity: %d Description: %s", stockType, stockCode, quantity, description);
-            cli.print(output);
-            return output;
         }
+
+        cli.print(output);
```

Comment:
Is this where SLAP is getting violated?

Reply:
i don't think so i just moved the code around because it made more sense this way